content/artifacts/highlight-active-artifact-planets/plugin.js

```diff
@@ -0,0 +1,168 @@
+// Display the planet with active artifact, specially for cannon.
+// When cannon first activated, only a thin circle can be seen around the planet.
+// As time goes by, a thick arc will appears, and it will extend longer and longer.
+// The thick arc become a circle when the cannon ready.
+
+import { 
+  BiomeNames
+} from "https://cdn.skypack.dev/@darkforest_eth/types";
+
+import {
+  coords
+} from 'https://plugins.zkga.me/utils/utils.js';
+
+import {
+  ArtifactType,
+  ArtifactTypeNames
+} from "https://cdn.skypack.dev/@darkforest_eth/types";
+
+const ARTIFACT_TYPES = Object.values(ArtifactType).filter((type) => type !== ArtifactType.Unknown).map((type) => ({ value: type, text: ArtifactTypeNames[type] }));
+
+
+function getPlayerColor(ethAddress) {
+  return df.getProcgenUtils().getPlayerColor(ethAddress);
+}
+
+
+function drawArc(ctx, planet, artifact, color){
+
+  let percent = 1;
+  if (artifact.artifactType === ArtifactType.PhotoidCannon) {
+    let timePass = Math.floor(Date.now() / 1000 - artifact.lastActivated);
+    percent = timePass / ui.gameManager.contractConstants.PHOTOID_ACTIVATION_DELAY;
+  }
+
+  const viewport = ui.getViewport();
+  ctx.strokeStyle = color;
+  ctx.lineWidth = 5;
+  if (!planet.location) {
+    return false;
+  }
+  const {x, y} = viewport.worldToCanvasCoords(planet.location.coords);
+  const range = planet.range * 0.01 * 20;
```

Comment:
I don't know if this is on purpose, but since you are using the planet.range variable a photoid cannon will have a very large circle (since it increases the range of the planet) and a planetary shield has a very small circle (sometimes smaller than the planet itself)
If you wanted an equal sized ring you could use something like this:
`const range = ui.getRadiusOfPlanetLevel(planet.planetLevel) * 4;`

Reply:
Thanks for the review, you right, it's done